ark/ops/ops_all_gather_msll.cc

```diff
@@ -14,40 +14,37 @@ constexpr int MAX_PEER_NUM = 7;
 // currently only support in single node
 MsllGatherFromPeersOp::MsllGatherFromPeersOp(
     const std::string &prec_type, Tensor *local_buf, Tensor *trans_region_local,
-    std::vector<Tensor *> remote_bufs, Tensor *trans_region_remote, int sid,
-    int rank, int npeers, size_t chunk_bytes, const std::string &name)
-    : Op(OP_GATHER_FROM_PEERS_MSLL, prec_type, {trans_region_remote},
-         {trans_region_local, local_buf}, {{rank, npeers, sid, chunk_bytes}},
-         name, &MsllGatherFromPeersConfigMap, -1, true) {
-    this->inputs.insert(this->inputs.end(), remote_bufs.begin(),
-                        remote_bufs.end());
-}
+    std::vector<Tensor *> remote_bufs, int sid, int rank, int npeers,
+    size_t stride, const std::string &name)
+    : Op(OP_GATHER_FROM_PEERS_MSLL, prec_type, remote_bufs,
+         {trans_region_local, local_buf}, {{rank, npeers, sid, stride}}, name,
+         &MsllGatherFromPeersConfigMap, -1, true) {}
 
 std::string MsllGatherFromPeersOp::function_name(const OpConfig &cfg) const {
     Tensor *dst_buff = this->outputs[0];
-    CHECK(dst_buff->is_sequential());
+    CHECK(dst_buff->ndims() == 1 || dst_buff->ndims() == 2);
```

Comment:
Any problems if we use `is_sequential()`?

Reply:
In this function, we pass two tensors as the output. The first one is the shared tensor, and the second is the original tensor. For the first one, we may get the tensor which shared by column. Then the buffer may not sequential. (For example, a tensor which shape is <8, 8>, after sharded by column, got two tensors which shape is <8, 4>)